Respawn/PostgresDbAdapter.cs

```diff
@@ -114,7 +114,55 @@ public string BuildDeleteCommandText(GraphBuilder graph)
             return builder.ToString();
         }
 
-        public string BuildReseedSql(IEnumerable<Table> tablesToDelete) => throw new System.NotImplementedException();
+        public string BuildReseedSql(IEnumerable<Table> tablesToDelete)
+        {
+            // Postgres has two sequence types, the "identity" column *type* and the serial *pseudo-type*.
+            // In the case of serial, some behind-the-scenes work is done to create a column of type int or
+            // bigint and then place a generated sequence behind it. Since both can be reseeded we can check
+            // for both, within the constraints of the tables the user wants to reset.
+            //
+            // This more complex implementation accommodates use cases where the user may have renamed their
+            // sequences in a manner where a regex parse would fail.
+
+            var builder = new StringBuilder();
+            builder.AppendLine(@"
+CREATE OR REPLACE FUNCTION pg_temp.reset_sequence(seq text) RETURNS void AS $$
+DECLARE
+BEGIN
+	/* ALTER SEQUENCE doesn't work with variables, so we construct a statement
+	 * and execute that instead.
+	 */
+	EXECUTE 'ALTER SEQUENCE ' || seq || ' RESTART;';
+END;
+$$ LANGUAGE plpgsql;
+");
+
+            var tableNames = string.Join(", ", tablesToDelete.Select(t => $"'{t.GetFullName(QuoteCharacter)}'"));
+
+            builder.AppendLine($@"
+WITH all_identities AS (
+	SELECT pg_get_serial_sequence(table_schema || '.' || table_name, column_name) AS sequence_name
+	FROM information_schema.columns
+	WHERE is_identity = 'YES'
+        AND '""' || table_schema || '"".""' || table_name || '""' IN ({tableNames}))");
+            builder.AppendLine("SELECT pg_temp.reset_sequence(i.sequence_name) FROM all_identities i;");
```

Comment:
I think you can get away with just one query for both "identity" and "serial", since `pg_get_serial_sequence` has results for both of them. Instead of filtering on `is_identity`, you can just check that `pg_get_serial_sequence` isn't returning null.
Aside from that, you'll want to `quote_ident` the schema and table names, in case there are some capital letters, spaces, or other weird characters in there.

```suggestion
            builder.AppendLine($@"
WITH all_identities AS (
	SELECT pg_get_serial_sequence(quote_ident(table_schema) || '.' || quote_ident(table_name), column_name) AS sequence_name
	FROM information_schema.columns
	WHERE pg_get_serial_sequence(quote_ident(table_schema) || '.' || quote_ident(table_name), column_name) IS NOT NULL
        AND '""' || table_schema || '"".""' || table_name || '""' IN ({tableNames}))
)
SELECT pg_temp.reset_sequence(i.sequence_name) FROM all_identities i;");
```